syndesis-maven-plugin/src/main/java/io/syndesis/maven/ExtractConnectorDescriptorsMojo.java

```diff
@@ -0,0 +1,284 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.descriptor.PluginDescriptor;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.DefaultProjectBuildingRequest;
+import org.apache.maven.project.ProjectBuildingRequest;
+import org.apache.maven.shared.artifact.resolve.ArtifactResolver;
+import org.apache.maven.shared.artifact.resolve.ArtifactResolverException;
+import org.apache.maven.shared.artifact.resolve.ArtifactResult;
+
+/**
+ * Simple maven plugin for creating a JSON file holding all connectors descriptors which are supported.
+ *
+ * @author roland
+ * @since 23.09.17
+ */
+@Mojo(name = "extract-connector-descriptors", defaultPhase = LifecyclePhase.GENERATE_RESOURCES)
+public class ExtractConnectorDescriptorsMojo extends AbstractMojo {
+
+    @Component
+    private ArtifactResolver artifactResolver;
+
+    @Parameter(defaultValue = "${session}", readonly = true, required = true)
+    private MavenSession session;
+
+    @Parameter(defaultValue = "${project.remoteArtifactRepositories}", readonly = true, required = true)
+    private List<ArtifactRepository> remoteRepositories;
+
+    @Parameter(defaultValue = "${project.build.outputDirectory}/META-INF/camel/camel-meta.json")
+    private String target;
+
+    @Override
+    @SuppressWarnings("PMD.EmptyCatchBlock")
+    public void execute() throws MojoExecutionException, MojoFailureException {
+
+        ArrayNode root = new ArrayNode(JsonNodeFactory.instance);
+
+        URLClassLoader classLoader = null;
+        try {
+            PluginDescriptor desc = (PluginDescriptor) getPluginContext().get("pluginDescriptor");
+            List<Artifact> artifacts = desc.getArtifacts();
+            ProjectBuildingRequest buildingRequest =
+                new DefaultProjectBuildingRequest(session.getProjectBuildingRequest());
+            buildingRequest.setRemoteRepositories(remoteRepositories);
+            for (Artifact artifact : artifacts) {
+                ArtifactResult result = artifactResolver.resolveArtifact(buildingRequest, artifact);
+                File jar = result.getArtifact().getFile();
+                classLoader = createClassLoader(jar);
```

Comment:
perhaps try-with-resources for the `classLoader`?

Reply:
I did try in the beginning but got ditched by the code checks (don't remember why anymore, but that was the quickest solution)